### DISCUSSION TITLE
How do I change the notebook output format/style in a Quarto manuscript project?

### DISCUSSION THREAD

**Discussion by mvuorre** [quarto-dev/quarto-cli#7444] [Q&A] [answered]
### Description

Hi, I am test driving [Quarto manuscripts](https://quarto.org/docs/manuscripts/). I would like to put my computations in a [notebook](https://quarto.org/docs/manuscripts/), but don't like the default html formatting of those. Is there a way to change how those notebooks look like?

For example, I created a bare bones manuscript project with `quarto create project manuscript quarto-manuscript-test` and this is what the notebook looks like:

<img width="799" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/ca15c2ee-0ff9-4be4-a368-21c34302f475">

How do I remove the "Ln [1]:" things and stop indenting code blocks? How do I get rid of the red around the code output? Also, things like turning `echo: false` for code execution have no effect. It appears that the notebooks produce `.ipynb` notebooks (see [here](https://github.com/mvuorre/quarto-manuscript-test/tree/main/docs)), but I would like to output regular html output files from R source code, like I do for regular Quarto html outputs. Can I do that?

Thanks for the exciting project so far :)

Example manuscript project: <https://mvuorre.github.io/quarto-manuscript-test/>
Example notebook output: <https://mvuorre.github.io/quarto-manuscript-test/test-notebook-preview.html>

**Comment by mcanouil** [comment, marked as answer]
It's just CSS.
Use your favourite browser inspect mode and tweak the CSS which you can then include back in your project.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7dc14b97-3c4c-41ae-9fbb-468484a955a9">

```yaml
include-in-header:
  - text: |
      <style>
      .quarto-notebook .cell-container .cell-decorator {
        display: none;
      }
      </style>
```

If you want more help, please share an actual reproducible example.

**Reply by dragonstyle** [reply]
(Note - that red border was an unintentional styling that I have removed and will disappear in subsequent pre-release builds)

**Reply by mvuorre** [reply]
I don't know what else <https://github.com/mvuorre/quarto-manuscript-test/tree/main> is but an "actual reproducible example". Would you like me to copy paste all the files' contents here @mcanouil?

The CSS trick works. Will it be possible to change the formatting/layout/execution of these notebooks similarly to what I can do with regular Quarto html documents? (e.g. `echo: false`, `theme: cosmo`, etc.)

[Here's](https://mvuorre.github.io/quarto-manuscript-test/test-notebook-preview.html) the current output with @mcanouil's CSS trick. I also tried a chunk option but that didn't work:

<pre>

---
title: "Test notebook"
include-in-header:
  - text: |
      <style>
      .quarto-notebook .cell-container .cell-decorator {
        display: none;
      }
      .quarto-notebook .cell .cell-output-stderr pre code, .quarto-notebook .cell .cell-output-stdout pre code {
        white-space: pre-wrap;
        border: none;
        overflow-wrap: anywhere;
      }
      </style>
---

Here's code for yall

```{r}
library(tidyverse)
```

echo: false doesn't work below

```{r}
#| echo: false
1 + 1
```
</pre>

<img width="787" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/f705c22b-84fb-4f77-950f-265e35cb2e6c">

**Reply by mcanouil** [reply]
You did not share the repository nor the steps to reproduce, only the rendered manuscript.
So yes, sharing the link of the repository with the code works.

Regarding how the notebook behaves, @dragonstyle will know more.

**Reply by mvuorre** [reply]
Come on dude, here's a screenshot of the original post: 
<img width="714" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/d82295f6-44fb-4ac5-b114-4d861bed3504">

**Reply by mcanouil** [reply]
That's not the repository links ...
You shared two pages from the GitHub Pages domain.
So please avoid "come on dude" comment when clearly it's on you.

And no, having to look through your repositories to find the match is not the way to do it.
Some users might end up on this discussion, so the links need to work and be explicit, as they might not know the correspondence between GitHub repository and GitHub Pages domains.

---

Could you properly format your last post to be legible, especially the code part?

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by dragonstyle** [reply]
I think the other thing I notice from your output that surprised me was the presence of the TOC in the notebooks (they are really intended to omit the TOC and lay themselves out like a notebook with cells). I will take a look at that as well.

Currently, the whole premise of notebooks within manuscript projects is to provide backing code and execution results as resources for the project (which is why we pretty specifically configure the output with the behavior that you're seeing). Can you say a bit more about why you want to control the output and when you see that being useful- we definitely could add options (or support the current options) but I want to be sure I understand what you're trying to do...

**Reply by mvuorre** [reply]
I see, that makes sense. I was mistakenly expecting notebooks (both the "manuscript" and additional notebooks) to behave like regular Quarto html documents. I guess I thought so because I could e.g. change the theme with html options as shown [here](https://quarto.org/docs/manuscripts/components.html#theme).

In other words, I thought I was creating a special kind of a Quarto website, where the landing page is the manuscript rendered in html, and with links to a. the manuscript in other formats (pdf, docx, etc.), and b. supplementary notebooks rendered in standard Quarto html. Had that been the case, I would have liked to e.g. use `|# echo: false` or any other formatting/execute/layout options in the notebooks. 

So a better option for my use-case is a standard Quarto website, but without any navigation features apart from a TOC on the landing page and below it links to supplementary html documents. Right?

Does that clarify it?

@mcanouil Here's something even better, a screenshot of the code 😜:
<img width="640" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/6cb17edc-bad2-44f1-8d75-353fa8145cd9">
